lib/std/zig/system.zig

```diff
@@ -1294,6 +1310,31 @@ fn elfInt(is_64: bool, need_bswap: bool, int_32: anytype, int_64: anytype) @Type
     }
 }
 
+fn detectAndroidApiLevel() !u32 {
+    var alloc_buf: [384]u8 = undefined;
+    var fba = std.heap.FixedBufferAllocator.init(&alloc_buf);
+    var child = std.process.Child.init(&.{ "/system/bin/getprop", "ro.build.version.sdk" }, fba.allocator());
+    // pass empty EnvMap, no allocator and deinit() required
+    child.env_map = &std.process.EnvMap.init(undefined);
+    child.stdin_behavior = .Ignore;
+    child.stdout_behavior = .Pipe;
+    child.stderr_behavior = .Ignore;
+
+    try child.spawn();
```

Comment:
Hmm. This (and the `try child.wait()` call below) will add a bunch of potential errors that [would need to](https://github.com/ziglang/zig/pull/24868/files#r2280453774) become part of `DetectError`. I'm not convinced that that's actually particularly helpful to a caller of `resolveTargetQuery`; I suggest marking `unreachable` those errors that there's no reason to expect would ever happen in this specific scenario, and mapping the rest to 1-2 higher level errors as appropriate that mention `Getprop` in the error name(s).